rust/core-lib/src/client.rs

```diff
@@ -74,6 +74,15 @@ impl Client {
         self.0.send_rpc_notification("available_languages", &json!({ "languages": languages }))
     }
 
+    pub fn content_changed(&self, view_id: ViewId) {
+        self.0.send_rpc_notification(
+            "content_changed",
+            &json!({
+                "view_id": view_id,
```

Comment:
Since we already capture the mtime of the file maybe we could also tell the frontend when the file has been changed? Not too relevant though, I suppose.

Reply:
How is the mtime retrieved from the front-end, is it too much additional work?

Also, is it relevant when notifying the user and asking how to proceed? The only use case I can think of is something along the lines of `The file has been changed outside of Xi {x minutes ago}` or `The file has been changed outside of Xi at {13:58 PM}`. But I don't know if that's good UX

Reply:
> How is the mtime retrieved from the front-end, is it too much additional work?

Xi already has that info, it uses that to determine if the file has been changed (arguably it should use checksumming for that)

> But I don't know if that's good UX

Neither do I, although I'd personally find it interesting

Reply:
I'm not sure if this is something the frontend can do much about. These notifications are going to get sent more or less immediately, and the frontend isn't supposed to really be touching the file-system or anything like that, so I'd be inclined to leave it out.